### ISSUE TITLE
OpeningFunctionBraceKernighanRitchie: bug/fixer conflict when return type is on new line

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#1809]
The `Generic.Functions.OpeningFunctionBraceKernighanRitchie` sniff goes into a fixer loop when the return type of a function declaration is on a new line.

This is caused by the sniff checking whether the last closing parentheses (either from the parameter declaration or from a closure use statement) is on the same line as the opening brace.

Example code:
```php
$x = $foo ? function ($foo)
    // comment
    : int {
    return 1;
} : $bar;
```

Looking at the code, the simplest fix would be to combine the `BraceOnNewLine` check with the `SpaceBeforeBrace` check instead of having both done separately.
In that case, we'd just check the first non-empty token before the open brace (in contrast to the last open parenthesis) and report on "0 space/ # space/ new line" found before open brace and fix it in one go.

@gsherwood I'd be happy to create the fix if you are ok with the proposed solution.

**Comment by gsherwood** [comment]
Just came across this again but can't generate any errors for that code sample any more. I think this may have changed due to all the other work going on with this sniff. Can you confirm?

**Comment by jrfnl** [comment]
@gsherwood Not at this moment, will need to do another run of the fixer conflict check and go through all the output in detail. It may be a while before I have more info. Please bear with me for this one.

**Comment by gsherwood** [comment]
Any news?

**Comment by jrfnl** [comment]
Not yet, sorry, been focused more on the new annotations over the last few months, than on getting the rest of the fixer conflicts sorted.
I'll try and do a new run this weekend and will report back.

**Comment by jrfnl** [comment]
@gsherwood Running the conflict check now. I had to make the fixes contained in PR #2029 and #2031 to get it to run again.
The updates around this "project" are now all over the place. Would it be helpful to have a dedicated issue for this to keep track of the status of the "project" ?

**Comment by jrfnl** [comment]
Ok, the initial run is now finished. I can confirm that commit ca108f2a75c45c739ae61b03de2f44c81b78e1fd fixed this issue.

**Comment by jrfnl** [comment]
@gsherwood Current status of the fixer conflict check:

PSR2: 3 files fail to fix
PSR12: 3 files fail to fix (same 3 as PSR2)
PEAR: 9 files fail to fix (5 of these would be solved by #1813)
Squiz: 43 files fail to fix
Zend: 3 files fail to fix

**Comment by gsherwood** [comment]
Thanks!

**Comment by jrfnl** [comment]
@gsherwood FYI: I've done a new fixer conflict check based on PHPCS 3.3.1.

Current status:

PSR2: 3 files fail to fix
PSR12: 3 files fail to fix (same 3 as PSR2)
PEAR: 9 files fail to fix (5 of these would be solved by #1813)
Squiz: 38 files fail to fix -> _down 5 due to bug fixes included in the 3.3.1 release_.
Zend: 3 files fail to fix

**Comment by jrfnl** [comment]
@gsherwood As you will have seen from the PRs I've been submitting, I've done a new fixer conflict check based on he current `master`.

Current status:
* PSR2: **2** files fail to fix (down by 1)
* PSR12: **2** files fail to fix (same 2 as PSR2)
* PEAR: **9** files fail to fix (5 of these would be solved by #1813, but need a different solution per the discussion there)
* Squiz: **59** (**up by 21 !!**) files fail to fix
    - PR #2266 brings this down by 2
    - PR #2277 brings this down by another whopping 16 :smile: 
    - PR #2279 takes care of another 9 of these
    - PR #2280 fixes yet one more
    - which brings the total down to **32** once these four have been merged (up one as one of the unit tests added exposes yet another fixer conflict bug).
    - and while PRs #2286, #2291 and #2292 don't definitely fix conflicts, they lessen a few conflicts, though additional fixes to other sniffs are still needed.
    - additionally PRs #2267, #2268, #2269, #2270 and #2272 (+ one more upcoming) are needed for the Squiz fixers to be able to even run on the CSS test case files without erroring out.
* Zend: **4** files fail to fix (up by one)
